tests/native/src/test/scala/epollcat/MonitorSuite.scala

```diff
@@ -0,0 +1,67 @@
+package epollcat
+
+import cats.effect.IO
+import cats.effect.Resource
+import epollcat.unsafe.EventPollingExecutorScheduler
+import epollcat.unsafe.EpollRuntime
+
+import scala.scalanative.unsafe._
+import scala.scalanative.unsigned._
+import scala.scalanative.posix.unistd._
+
+class MonitorSuite extends EpollcatSuite {
+
+  val zoneResource: Resource[IO, Zone] = Resource.make(IO(Zone.open()))(z => IO(z.close()))
+
+  val pipeResource: Resource[IO, (Int, Int)] = Resource.make {
+    zoneResource.use { implicit zone =>
+      val fildes = alloc[CInt](2)
+      if (pipe(fildes) != 0) {
+        IO.raiseError(new Exception("Failed to create pipe"))
+      } else {
+        IO((fildes(0), fildes(1)))
+      }
+    }
+  } {
+    case (a, b) =>
+      IO {
+        println("CLOSING PIPES")
+        close(a)
+        close(b)
+        ()
+      }
+  }
+
+  test("compile") {
+    val scheduler = EpollRuntime.global.scheduler.asInstanceOf[EventPollingExecutorScheduler]
+
+    pipeResource.use {
+      case (r, w) =>
+        IO.fromFuture {
+          IO {
+            println("TEST STARTS")
+            val promise = scala.concurrent.Promise[Unit]()
+            val byte = 10.toByte
+            scheduler.monitor(r, reads = true, writes = false)(
```

Comment:
Is the test or the test suite hanging? Note that the test suite will hang until all monitors are de-registered.

Reply:
You're right, in fact I had `poll.stop()` in the test I'm trying to port from scala-native-loop. stopping in the callback makes the test pass...